app/controllers/apis/api_controller.rb

```diff
@@ -0,0 +1,43 @@
+# apis controller
+module Apis
+  class ApiController < ApplicationController
+    skip_before_action :authenticate_user!
+    protect_from_forgery with: :null_session
+    before_action :set_params_hash, if: :request_is_get?
+
+    def request_is_get?
+      request.method == "GET" && !params["params"].nil?
+    end
+
+    def request_is_valid_post?
+      request.method == "POST" && !except_params.nil?
```

Comment:
Since `except_params` returns boolean value, `!except_params.nil?` will always be true.